src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h

```diff
@@ -0,0 +1,136 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2017 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+#ifndef POSIX_EAGER_IQUEUE_SEND_H_INCLUDED
+#define POSIX_EAGER_IQUEUE_SEND_H_INCLUDED
+
+#include "iqueue_impl.h"
+
+MPL_STATIC_INLINE_PREFIX MPIDI_POSIX_EAGER_IQUEUE_cell_t
+    * MPIDI_POSIX_EAGER_IQUEUE_new_cell(MPIDI_POSIX_EAGER_IQUEUE_transport_t * transport)
+{
+    int i;
+    for (i = 0; i < transport->num_cells; i++) {
+        MPIDI_POSIX_EAGER_IQUEUE_cell_t *cell = MPIDI_POSIX_EAGER_IQUEUE_THIS_CELL(transport, i);
+        if (cell->type == MPIDI_POSIX_EAGER_IQUEUE_CELL_TYPE_NULL) {
+            return cell;
+        }
+    }
+    return NULL;
+}
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_POSIX_eager_send
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int
+MPIDI_POSIX_eager_send(int grank,
+                       MPIDI_POSIX_am_header_t ** msg_hdr,
+                       struct iovec **iov, size_t * iov_num, int is_blocking)
+{
+    MPIDI_POSIX_EAGER_IQUEUE_transport_t *transport;
```

Comment:
Looks like we don't have `MPIR_FUNC_VERBOSE_STATE_DECL` set in all `fbox` and `iqueue` internal routines. Is it intentional ?

Reply:
Probably not. I'll add them.